docs/docs/development/studio-setup.md

```diff
@@ -0,0 +1,80 @@
+
+- Width
+- Height
+- X
+- Y
+- Rotation
+- Rotation X
+- Rotation Y
+
+All X/Y/Width/Height values are in "key unit" units, with a 100th of a key unit resolution. Rotations are in degrees, with a 100th of a degree resolution. Due to the nature of devicetree, all values are stored in the devicetree without decimals as hundredths of a value, e.g. one key unit is represented as `100`, twenty degrees is represented as `2000`, etc.
+
+Here is an example physical layout for a 2x2 macropad:
+
+```
+    macropad_physical_layout: macropad_physical_layout {
+        compatible = "zmk,physical-layout";
+        display-name = "Macro Pad";
+
+        keys  //                     w   h    x    y     rot    rx    ry
+            = <&key_physical_attrs 100 100    0    0       0     0     0>
+            , <&key_physical_attrs 100 100  100    0       0     0     0>
+            , <&key_physical_attrs 100 100    0  100       0     0     0>
+            , <&key_physical_attrs 100 100  100  100       0     0     0>
+            ;
+    };
+```
+
+# Position Map
+
+When switching between layouts with ZMK Studio, the keymap of the previously selected layout is used to populate the keymap in the new layout. To determine which keymap entry maps to which entry in the new layout, keys between the two layouts are matched that share the exact same physical attributes. However, in some cases, keys between layouts are not in exactly the same position.
+
+For those cases, a position map is used. The position map includes a sequence for every relevant layout, and the corresponding entries in the `positions` will be used to determine the mapping.
+
+For example, the following position map correctly maps the 5-column and 6-column Corne keymap layouts.
+
+```
+    foostan_corne_position_map {
+        compatible = "zmk,physical-layout-position-map";
+
+        complete;
+
+        twelve {
+            physical-layout = <&foostan_corne_6col_layout>;
+            positions
+                = <    1  2  3  4  5  6  7  8  9 10   >
+                , <   13 14 15 16 17 18 19 20 21 22   >
+                , <   25 26 27 28 29 30 31 32 33 34   >
+                , <         36 37 38 39 40 41               >;
+        };
+
+        ten {
+            physical-layout = <&foostan_corne_5col_layout>;
+            positions
+                = < 0  1  2  3  4  5  6  7  8  9>
+                , <10 11 12 13 14 15 16 17 18 19>
+                , <20 21 22 23 24 25 26 27 28 29>
+                ,       <30 31 32 33 34 35>;
+        };
+    };
+```
+
+The first entries in the two mappings have values `1` and `0` respectively, which means that the zero-th entry in the 5-column will map to the first entry in the 6-column layout, the second entries show that the second 6-column keymap entry corresponds to the first 5-column entry, etc.
```

Comment:
I think this is missing documenting the `complete;` property.

Reply:
Added documentation above the example.